api.planx.uk/inviteToPay/createPaymentSendEvents.ts

```diff
@@ -0,0 +1,68 @@
+import { NextFunction, Request, Response } from 'express';
+import { createScheduledEvent } from "../hasura/metadata";
+import { _admin as $admin } from '../client';
+import { getMostRecentPublishedFlow } from '../helpers';
+
+type Destination = "bops" | "uniform" | "email";
+
+interface CombinedResponse {
+  bops?: Record<string, string>;
+  uniform?: Record<string, string>;
+  email?: Record<string, string>;
+}
+
+// Create "One-off Scheduled Events" in Hasura when a payment request is paid
+const createPaymentSendEvents = async (req: Request, res: Response, next: NextFunction): 
+  Promise<NextFunction | Response | void> => {
+  try {
+    const now = new Date();
+    const combinedResponse: CombinedResponse = {};
+
+    // TODO Get this session's published flow data, find it's Send component, determine which "destinations" we need to queue up events for
+    const session = await $admin.getSessionById(req.params.sessionId);
+    const publishedFlowData = await getMostRecentPublishedFlow(session.flowId);
+    const destinations: Destination[] = [];
+
+    // TODO Determine which local authority it should send to based on team & destination    
+
+    const eventPayload = { sessionId: req.params.sessionId };
+    if ("bops" in destinations) {
```

Comment:
I think this should be:

```suggestion
    if (destinations.includes("bops")) {
```
and similar for other `if` statements below.

The [`in` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) looks for properties in an object rather than elements in an array (well it works on the index of the array rather than the value).

Sorry if this is actually just pseudo-code while this is still in draft. I was just being a little nosy :nose:

Reply:
you're right! this is just pseudo code still to stub out general logic, will update and properly run/test soon :sweat_smile: